dango/types/src/oracle/pyth.rs

```diff
@@ -0,0 +1,235 @@
+                                publish_time: price_feed_message.publish_time,
+                            },
+                        ),
+                        _ => bail!("invalid message"),
+                    };
+
+                    Ok(price_feed)
+                })
+                .collect::<anyhow::Result<Vec<_>>>()?;
+
+            (parsed_vaa, feeds)
+        } else {
+            let vaa = WormholeVAA::new(bytes)?;
+            let batch_attestation =
+                BatchPriceAttestation::deserialize(&vaa.payload[..]).map_err(|err| {
+                    anyhow::anyhow!("Failed to deserialize batch attestation: {:?}", err)
+                })?;
+
+            let feeds: Vec<_> = batch_attestation
+                .price_attestations
+                .into_iter()
+                .map(|price_attestation| match price_attestation.status {
+                    PriceStatus::Trading => PriceFeed::new(
+                        PriceIdentifier::new(price_attestation.price_id.to_bytes()),
+                        Price {
+                            price: price_attestation.price,
+                            conf: price_attestation.conf,
+                            expo: price_attestation.expo,
+                            publish_time: price_attestation.publish_time,
+                        },
+                        Price {
+                            price: price_attestation.ema_price,
+                            conf: price_attestation.ema_conf,
+                            expo: price_attestation.expo,
+                            publish_time: price_attestation.publish_time,
+                        },
+                    ),
+                    _ => PriceFeed::new(
+                        PriceIdentifier::new(price_attestation.price_id.to_bytes()),
+                        Price {
+                            price: price_attestation.prev_price,
+                            conf: price_attestation.prev_conf,
+                            expo: price_attestation.expo,
+                            publish_time: price_attestation.prev_publish_time,
+                        },
+                        Price {
+                            price: price_attestation.ema_price,
+                            conf: price_attestation.ema_conf,
+                            expo: price_attestation.expo,
+                            publish_time: price_attestation.prev_publish_time,
+                        },
+                    ),
+                })
+                .collect();
+
+            (vaa, feeds)
+        };
+
+        Ok(PythVaa { vaa, feeds })
+    }
+}
+
+impl FromStr for PythVaa {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> anyhow::Result<Self> {
+        PythVaa::new(Binary::from_str(s)?.into_inner())
+    }
+}
+
+pub struct PythVAAVisitor;
+
+impl<'de> Visitor<'de> for PythVAAVisitor {
+    type Value = PythVaa;
+
+    fn expecting(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
```

Comment:
import `std::fmt` and only use `fmt::Formatter` here (no `std::`). this is how we do it in every other file in the codebase, so keep consistent.